src/Stratis.Bitcoin.Tests.Common/TestBase.cs

```diff
@@ -25,31 +25,10 @@ public TestBase(Network network)
 
         public static string AssureEmptyDir(string dir)
         {
-            int deleteAttempts = 0;
-            while (deleteAttempts < 50)
-            {
-                if (Directory.Exists(dir))
-                {
-                    try
-                    {
-                        Directory.Delete(dir, true);
-                        break;
-                    }
-                    catch
-                    {
-                        deleteAttempts++;
-                        Thread.Sleep(200);
-                    }
-                }
-                else
-                    break;
-            }
-
-            if (deleteAttempts >= 50)
-                throw new Exception(string.Format("The test folder: {0} could not be deleted.", dir));
-
-            Directory.CreateDirectory(dir);
-            return dir;
+            long ticks = DateTime.UtcNow.Ticks;
+            string uniqueDirName = dir + ticks.ToString();
```

Comment:
ticks is not very readable. It'd be hard to quickly see the results of the tests you ran a couple of days ago.
I prefer `$"{dir}-{DateTime.UtcNow.ToString("dd/MM/yyyyTHH:mm:ss.fff")}"` which will give you folders like _forldername_-15/08/2018T15:40:43.144

Reply:
+1, but you may need to use underscores etc in lieu of characters that are invalid in folder names.

Reply:
accept